solutions/cses-1736.mdx

```diff
@@ -0,0 +1,158 @@
+---
+id: cses-1736
+source: CSES
+title: Polynomial Queries
+author: Albert Ye
+---
+
+## Explanation
+
+Despite the strange operations we can still use a segment tree with lazy propagation
+to solve this problem.
+
+The range queries are vanilla lazy propagation, so we should focus more on the updates.
+
+We can split the "update" operation into two operations. One would be a basic range addition,
+and the other would be an incremental addition. Therefore, for each index $a_i$, we will need
+to keep track of two points: the total offset $s_i$ given only the offset sum (without increments),
```

Comment:
"Therefore, for each index $a_i$, we will need
to keep track of two points" a bit unclear (?)

Did you mean "for each segment (in the segment tree), keep track of the ..."

Reply:
yeah that's a bit better worded i'm very scuffed at this lol